### PR TITLE
MS Transferor code re-factoring and clean-up according to proposal specs

### PR DESCRIPTION
**Pull request by vkuznet**

### CODE REVIEW

**Comment by vkuznet** [comment]
Code re-factoring and clean-up of intermediate changes from #9140

**Comment by amaltaro** [comment]
@vkuznet please have a look at the report html provided by the jenkins setup and try to clean some of those (usually errors and warnings deserve attention).

Messages like
` W1201, line 402 in requestsInfo_old: Specify string format arguments as logging function parameters`

means that your logging line should be written as:
```
logger.debug("request: %s, campaignConfig: %s", reqName, campaignConfig)
```

instead of
```
logger.debug("request: %s, campaignConfig: %s" % (reqName, campaignConfig))
```

Please also describe these changes in the first comment box.

**Comment by vkuznet** [comment]
Alan, since you merged #9217  it created a conflict with this PR. How do you want to resolve it? I already expressed my opinion in PR#9217.

**Comment by amaltaro** [comment]
Can you please rebase and resolve this conflict?

**Comment by vkuznet** [comment]
Alan, can you trigger Pylint tests again, it reports suggestions for code which I already fixed/no longer exists.

**Comment by amaltaro** [comment]
@vkuznet I haven't reviewed this pull request yet, but can you squash your changes into 2 commits please (1 for src and 1 for test)? just in case I get it merged by tomorrow morning.

**Comment by vkuznet** [comment]
@amaltaro , Alan I tested this PR in my VM and it works. The entire data flow works correctly ,i.e. MS Transferor and MSTransferorMonit. I also cleaned up all pylints errors and cleaned up the code base itself. I put in place the unified version which I had originally and it works too, i.e. it fetches aux info for given request and produce information which we can rely on, e.g. number of events, lumis, sites, etc. Once this will be merged we can start working on individual PRs to tune-up unified logic, do transfers, etc. Feel free to make another review round and you may include this PR into June release.

**Comment by vkuznet** [comment]
and, all commits a squashed and I used internally the record structures (option C) from your gist.

**Comment by amaltaro** [comment]
Awesome, thanks!
For future cases, please keep pylint + unit tests fixes in a separate commit, it makes it easier to review.

I've have made some deployment changes here: https://github.com/dmwm/deployment/pull/776
I still have to test and see whether we're missing anything deployment-wise.

File: src/python/WMCore/MicroService/Unified/Transferor.py [hunk 1]
```diff
@@ -131,37 +67,42 @@ def monit(self, req_status='staging'):
...
+            for reqName in requests:
+                req = {'name':reqName, 'reqStatus': reqStatus}
                 # get transfer IDs
                 tids = self.getTransferIDs()
                 # get transfer status
                 transferStatuses = self.getTransferStatuses(tids)
                 # get campaing and unified configuration
-                campain = self.requestCampaign(req)
-                conf = self.requestConfiguration(req)
+                campaign = self.requestCampaign(reqName)
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/Transferor.py]: a cache needs to be put in place here. 10min should suffice.

File: src/python/WMCore/MicroService/Unified/Transferor.py [hunk 2]
```diff
@@ -170,229 +111,169 @@ def transferor(self, req_status='assigned'):
         https://github.com/dmwm/WMCore/wiki/ReqMgr2-MicroService-Transferor
         """
         try:
+            # get requests from ReqMgr2 data-service for given statue
+            requestSpecs = self.svc.reqmgr.getRequestByStatus([reqStatus], detail=True)
+            requestRecords = [requestRecord(r, reqStatus) for r in requestSpecs]
+            self.logger.debug('### monit found %s requests in %s state', len(requestRecords), reqStatus)
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/Transferor.py]: monit or transferor?

File: src/python/WMCore/MicroService/Unified/Transferor.py [hunk 3]
```diff
@@ -170,229 +111,169 @@ def transferor(self, req_status='assigned'):
...
+    def transferRequest(self, req):
         "Send request to Phedex and return status of request subscription"
-	datasets = rdict.get('datasets', [])
-	sites = rdict.get('sites', [])
+        datasets = req.get('datasets', [])
+        sites = req.get('sites', [])
         if datasets and sites:
-	    subscription = PhEDExSubscription(datasets, sites, self.group)
-	    self.logger.debug("### add subscription %s for request %s" % (subscription, request))
+            subscription = PhEDExSubscription(datasets, sites, self.group)
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/Transferor.py]: Are we able to subscribe blocks as well? Or only datasets?

File: src/python/WMCore/MicroService/Unified/Transferor.py [hunk 4]
```diff
@@ -170,229 +111,169 @@ def transferor(self, req_status='assigned'):
...
         https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
-	{"wf_A": {"timestamp": 0000
-		  "primary": {"dset_1": ["list of transfer ids"]},
-		  "secondary": {"PU_dset_1": ["list of transfer ids"]},
-	 "wf_B": {"timestamp": 0000
-		  "primary": {"dset_1": ["list of transfer ids"],
-			      "parent_dset_1": ["list of transfer ids"]},
-		  "secondary": {"PU_dset_1": ["list of transfer ids"],
-				"PU_dset_2": ["list of transfer ids"]},
+        {"wf_A": {"timestamp": 0000
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/Transferor.py]: this data structure is no longer going to be used. We decided to adopt option C from the gist.

**Review by amaltaro — COMMENTED** [review]
Valentin, I have put several comments in place. If you decide to address them before I wake up tomorrow, please do so in a second commit. Otherwise, we can likely proceed with this PR and then apply some fine tuning in a later stage (or I can get it in tomorrow morning).

**Comment by vkuznet** [comment]
Alan,
these are very good comments and they show that I'm lack of some knowledge of
DMWM. I don't have time tonight and I think it should be better of if I'll
address your suggestions separately as I don't want to rush into them in a
morning and hold you off for making a release cut.
Valentin.

**Pull request merged**